charts/redpanda/templates/_helpers.tpl

```diff
@@ -305,6 +305,39 @@ Generate configuration needed for rpk
   {{- $result -}}
 {{- end -}}
 
+{{/*
+Returns the value of "resources.cpu.cores" in millicores. And ensures CPU units
+are using known suffix (really only "m") or no suffix at all.
+*/}}
+{{- define "redpanda-cores-in-millis" -}}
+  {{- $cores := .Values.resources.cpu.cores | toString -}}
+  {{- $coresSuffix := regexReplaceAll "^[0-9.]+(.*)" $cores "${1}" -}}
+  {{- if eq $coresSuffix "m" -}}
+    {{- trimSuffix $coresSuffix .Values.resources.cpu.cores -}}
+  {{- else -}}
+    {{- if eq $coresSuffix "" -}}
+      {{- mulf 1000.0 ($cores | float64) -}}
+    {{- else -}}
+      {{- printf "Unrecognized CPU unit '%s'" $coresSuffix | fail -}}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Returns the SMP CPU count in whole cores, with minimum of 1, and sets
+"resources.cpu.overprovisioned: true" when the "resources.cpu.cores" is less
+than 1 core.
+*/}}
+{{- define "redpanda-smp" -}}
+  {{- $coresInMillies := include "redpanda-cores-in-millis" . | int -}}
+  {{- if lt $coresInMillies 1000 -}}
+    {{- $_ := set $.Values.resources.cpu "overprovisioned" true -}}
+    {{- 1 -}}
```

Comment:
This could just be
```suggestion
1
```
if you wanted

Reply:
:) I was debating on it, but wanted to keep the leading space padding